### PR TITLE
Fix case which flag app-name has double quotes and add examples

### LINKED ISSUE #619: Simplify sysl codegen command
### Current Sysl Version
`sysl codegen --transform=one.sysl --grammar=grammar.go.gen --start=goFile --out=gen/ --app-name=foo model.sysl`
This currently needs to be repeated for every transform for every app

A better approach would be to use a config issue or something:
`sysl codegen --config=config.yaml --app-name=simple out=gen/ model.sysl`

Which can then infer all the `server` and `client` (possibly by putting the required transforms in server/ and client/ dirs) and having the required `grammar.go.gen` somewhere

### Attempted Solutions
Currently a long and tedious `makefile` is used
EDIT:
I don't like the idea of an environment variable

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #619 .

Changes proposed in this pull request:
- Fix case which flag app-name has double quotes and add examples

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: pkg/syslutil/cmd_config.go [hunk 1]
```diff
@@ -12,8 +13,12 @@ func ReadCMDFlags(configPath string) ([]string, error) {
 	if ferr != nil {
 		return flags, ferr
 	}
-	str := string(data)
-	flags = strings.Fields(strings.TrimSpace(str))
+	re := regexp.MustCompile("(\\S+\"[^\"]+\")|\\S+")
```
Comment (anzdaddy) [inline pkg/syslutil/cmd_config.go]:
```suggestion
	re := regexp.MustCompile(`(\S+"[^"]+")|\S+`)
```

File: pkg/syslutil/cmd_config.go [hunk 2]
```diff
@@ -12,8 +13,12 @@ func ReadCMDFlags(configPath string) ([]string, error) {
...
+	flags = re.FindAllString(strings.TrimSpace(string(data)), -1)
```
Comment (anzdaddy) [inline pkg/syslutil/cmd_config.go]: Initialise `flags` here.

File: pkg/syslutil/cmd_config.go [hunk 3]
```diff
@@ -12,8 +13,12 @@ func ReadCMDFlags(configPath string) ([]string, error) {
...
+
+	for i, flag := range flags {
+		flags[i] = strings.ReplaceAll(flag, "\"", "")
```
Comment (anzdaddy) [inline pkg/syslutil/cmd_config.go]: This is dangerous. Sometimes you want to pass a double-quote as part of a parameter. Better to use a regex to match a common flag-or-arg pattern: `^(-+[^=]+=)?"?(.*?)"?$`.

File: pkg/syslutil/cmd_config.go (hunk 1 above)
Comment (ericzhang6222) [reply pkg/syslutil/cmd_config.go]: Fixed.

File: pkg/syslutil/cmd_config.go (hunk 2 above)
Comment (ericzhang6222) [reply pkg/syslutil/cmd_config.go]: Fixed.

File: pkg/syslutil/cmd_config.go (hunk 3 above)
Comment (ericzhang6222) [reply pkg/syslutil/cmd_config.go]: Checking

Comment (ericzhang6222) [reply pkg/syslutil/cmd_config.go]: Add a new pattern to parse flags config: `(-+[^=\s\n]+)?((=?"[^=\n]+")|(=?[^=\s\n]+))?\s*`

**Pull request closed without merging**